src/datepicker/datepicker.js

```diff
@@ -219,7 +219,16 @@ export default class Datepicker extends React.Component<
         }
       }
     } else {
-      const date = new Date(inputValue);
+      let date = new Date(inputValue);
```

Comment:
shouldn't we use datefn's `parse` here instead?

Reply:
That's a good question. parse requires string format. In theory that should be `dd/MM/YYYY` as it's default for datepicker.

Reply:
we could do something like `this.props.formatString || 'yyyy/MM/dd'` (I think this is the default, based on https://github.com/uber/baseweb/blob/master/src/datepicker/datepicker.js#L115)

Reply:
Also, I think, that it doesn't really make sense. We're handling the parsing of date further, and keep it as default. 
As I can see, in range it's done same way.

Reply:
for the range, we do this:

```
if (formatString) {
  startDate = parse(left, this.normalizeDashes(formatString), new Date());
  endDate = parse(right, this.normalizeDashes(formatString), new Date());
}
```

do you think it would make sense for the single select to be something like this

```js
const dateString = this.normalizeDashes(inputValue);
let date = new Date(datestring)

if (formatString) {
  date = parse(dateString, this.normalizeDashes(formatString), new Date());
}

      isValid(date) &&
        this.props.onChange &&
        this.props.onChange({
          date,
        });
```

Reply:
normalizeDashes does makes sense. But the if condition is a missing the case of early-parsing.
If you parse the date `11.12.2` it would be parsed as `11.12.0002` and will be replaced in input. So, the strict lenght check is needed.

Reply:
understood, that makes sense - thanks Vladimir